luigi/contrib/ftp.py

```diff
@@ -127,10 +127,14 @@ def _sftp_exists(self, path, mtime):
         return exists
 
     def _ftp_exists(self, path, mtime):
+        path_parts = path.split('/')
+        path = '/'.join(path_parts[:-1])
+        fn = path_parts[-1]
```

Comment:
Ah, I read `if fn in files` as `for fn in files`. Sorry, still morning over here ^^